hamilton/plugins/jupyter_magic.py

```diff
@@ -163,95 +194,160 @@ def determine_notebook_type() -> str:
-            print("  -c, --config: JSON config string, or variable name containing config to use.")
-            print("  -r, --rebuild-drivers: Flag to rebuild drivers.")
-            print("  -d, --display: Flag to visualize dataflow.")
-            print("  -v, --verbosity: of standard output. 0 to hide. 1 is normal, default.")
-            return  # Exit early
-        if not hasattr(self, "notebook_env"):
-            # doing this so I don't have to deal with the constructor
-            self.notebook_env = determine_notebook_type()
         # shell.ex() is equivalent to exec(), but in the user namespace (i.e. notebook context).
         # This allows imports and functions defined in the magic cell %%cell_to_module to be
         # directly accessed from the notebook
         self.shell.ex(cell)
 
         args = parse_argstring(self.cell_to_module, line)  # specify how to parse by passing method
-        module_name = "jupyter_module" if args.module_name is None else args.module_name
-
-        display_config = {}
-        if args.config:
-            if args.config in self.shell.user_ns:
-                display_config = self.shell.user_ns[args.config]
+        module_name = args.module_name
+        builder = self.shell.user_ns[args.builder] if args.builder else driver.Builder()
+        inputs = self.shell.user_ns[args.inputs] if args.inputs else {}
+        overrides = self.shell.user_ns[args.overrides] if args.overrides else {}
+        display_config = (
+            self.shell.user_ns[args.display] if args.display not in [True, None] else {}
+        )
+
+        if args.write_to_file:
+            if isinstance(args.write_to_file, str):
+                file_path = Path(args.write_to_file)
             else:
-                try:
-                    if args.config.startswith("'") or args.config.startswith('"'):
-                        # strip quotes if present
-                        args.config = args.config[1:-1]
-                    display_config = json.loads(args.config)
-                except json.JSONDecodeError:
-                    print("Failed to parse config as JSON. Please ensure it's a valid JSON string:")
-                    print(args.config)
-
-        module_object = create_module(cell, module_name)
-
-        # shell.push() assign a variable in the notebook. The dictionary keys are variable name
+                file_path = Path(f"{module_name}.py")
+            file_path.write_text(cell)
+
+        # create modules and build driver
+        module_object = ad_hoc_utils.module_from_source(cell, module_name)
         self.shell.push({module_name: module_object})
+        base_dr = builder.build()  # easier to rebuild a Driver than messing with Builder
+        dr = rebuild_driver(dr=base_dr, modules=[*base_dr.graph_modules, module_object])
 
-        # shell.user_ns is a dictionary of all variables in the notebook
-        # rebuild drivers that use this module
-        if args.rebuild_drivers:
```

Comment:
I think this should be left in to keep things backwards compatible.

Reply:
Wouldn't it be confusing to preserve the arg but changing it's behavior?

Regarding migration (this and `--module_name`), a user of the old magic would get:
```python
%%cell_to_module -m joke -r
def reply(joke_prompt: str) -> str:
    _, _, right = joke_prompt.partition("? ")
    return f"{right} who?"
```
Error
```console
UsageError: unrecognized arguments: -m -r
```
This would prompt them to do `%%cell_to_module?` and get the updated docs.

As a change, I could allow the old args and do:
```python
if any(args.rebuild_driver, args.module_name, ...):
   raise ValueError("Arguments ... are deprecated. Use `%%cell_to_module?` for updated documentation")
```